### PR TITLE
Integrate all the "select" function to a single one

### PR DESCRIPTION
**Pull request by ThisIsClark**
**What this PR does / why we need it**:
Nowadays, each type of source has its own select function, like"SelectFileShares", "SelectFileShareSnapshots", "SelectPools", "SelectDocks", and so on.
Each of the select function almost do the same things, and the only difference is that the data type which they selected is different.
I write a function to select all the type of source to avoid shotgun surgery when we need to maintain the select function.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: 

**Special notes for your reviewer**:
This pr integrate all the "select" function(like SelectPools, SelectDocks, SelectProfiles...) to one.
And next step, I want to integrate all the "sort" function(like SortPools, SortDocks, SortProfiles).

**Release note**:

```
Unit Test:
......
=== RUN   TestFilterAndSort
--- PASS: TestFilterAndSort (0.00s)
=== RUN   TestListFileSharesWithFilter
--- PASS: TestListFileSharesWithFilter (0.00s)
=== RUN   TestListFileShareSnapshotsWithFilter
--- PASS: TestListFileShareSnapshotsWithFilter (0.00s)
=== RUN   TestListDocksWithFilter
--- PASS: TestListDocksWithFilter (0.00s)
=== RUN   TestListPoolsWithFilter
--- PASS: TestListPoolsWithFilter (0.00s)
=== RUN   TestListProfilesWithFilter
--- PASS: TestListProfilesWithFilter (0.00s)
=== RUN   TestListVolumesWithFilter
--- PASS: TestListVolumesWithFilter (0.00s)
=== RUN   TestListVolumeAttachmentsWithFilter
--- PASS: TestListVolumeAttachmentsWithFilter (0.00s)
=== RUN   TestListVolumeSnapshotsWithFilter
--- PASS: TestListVolumeSnapshotsWithFilter (0.00s)
PASS
coverage: 32.2% of statements
ok      github.com/opensds/opensds/pkg/db/drivers/etcd  (cached)        coverage: 32.2% of statements

Test with pool list:
root@node01:~/gopath/src/github.com/opensds/opensds (development) # ./build/out/bin/osdsctl pool list
WARNING: OPENSDS_ENDPOINT is not specified, use default(http://localhost:50040)
WARNING: Not found Env OPENSDS_AUTH_STRATEGY, use default(noauth)
+--------------------------------------+-------------------------+-------------+-----------+---------------+--------------+
| Id                                   | Name                    | Description | Status    | TotalCapacity | FreeCapacity |
+--------------------------------------+-------------------------+-------------+-----------+---------------+--------------+
| 47d06aef-afe0-5f34-90c1-66c355662a9b | opensds-volumes-default |             | available | 20            | 20           |
| 157e862c-7604-5d1b-b80a-37559a88e65d | opensds-files-default   |             | available | 20            | 20           |
+--------------------------------------+-------------------------+-------------+-----------+---------------+--------------+
root@node01:~/gopath/src/github.com/opensds/opensds (development) # ./build/out/bin/osdsctl pool list --sortKey name --sortDir asc
WARNING: OPENSDS_ENDPOINT is not specified, use default(http://localhost:50040)
WARNING: Not found Env OPENSDS_AUTH_STRATEGY, use default(noauth)
+--------------------------------------+-------------------------+-------------+-----------+---------------+--------------+
| Id                                   | Name                    | Description | Status    | TotalCapacity | FreeCapacity |
+--------------------------------------+-------------------------+-------------+-----------+---------------+--------------+
| 157e862c-7604-5d1b-b80a-37559a88e65d | opensds-files-default   |             | available | 20            | 20           |
| 47d06aef-afe0-5f34-90c1-66c355662a9b | opensds-volumes-default |             | available | 20            | 20           |
+--------------------------------------+-------------------------+-------------+-----------+---------------+--------------+
root@node01:~/gopath/src/github.com/opensds/opensds (development) # ./build/out/bin/osdsctl pool list --sortKey name --sortDir desc
WARNING: OPENSDS_ENDPOINT is not specified, use default(http://localhost:50040)
WARNING: Not found Env OPENSDS_AUTH_STRATEGY, use default(noauth)
+--------------------------------------+-------------------------+-------------+-----------+---------------+--------------+
| Id                                   | Name                    | Description | Status    | TotalCapacity | FreeCapacity |
+--------------------------------------+-------------------------+-------------+-----------+---------------+--------------+
| 47d06aef-afe0-5f34-90c1-66c355662a9b | opensds-volumes-default |             | available | 20            | 20           |
| 157e862c-7604-5d1b-b80a-37559a88e65d | opensds-files-default   |             | available | 20            | 20           |
+--------------------------------------+-------------------------+-------------+-----------+---------------+--------------+
root@node01:~/gopath/src/github.com/opensds/opensds (development) # ./build/out/bin/osdsctl pool list --name opensds-volumes-default
WARNING: OPENSDS_ENDPOINT is not specified, use default(http://localhost:50040)
WARNING: Not found Env OPENSDS_AUTH_STRATEGY, use default(noauth)
+--------------------------------------+-------------------------+-------------+-----------+---------------+--------------+
| Id                                   | Name                    | Description | Status    | TotalCapacity | FreeCapacity |
+--------------------------------------+-------------------------+-------------+-----------+---------------+--------------+
| 47d06aef-afe0-5f34-90c1-66c355662a9b | opensds-volumes-default |             | available | 20            | 20           |
+--------------------------------------+-------------------------+-------------+-----------+---------------+--------------+
root@node01:~/gopath/src/github.com/opensds/opensds (development) # ./build/out/bin/osdsctl pool list --id 157e862c-7604-5d1b-b80a-37559a88e65d
WARNING: OPENSDS_ENDPOINT is not specified, use default(http://localhost:50040)
WARNING: Not found Env OPENSDS_AUTH_STRATEGY, use default(noauth)
+--------------------------------------+-----------------------+-------------+-----------+---------------+--------------+
| Id                                   | Name                  | Description | Status    | TotalCapacity | FreeCapacity |
+--------------------------------------+-----------------------+-------------+-----------+---------------+--------------+
| 157e862c-7604-5d1b-b80a-37559a88e65d | opensds-files-default |             | available | 20            | 20           |
+--------------------------------------+-----------------------+-------------+-----------+---------------+--------------+
root@node01:~/gopath/src/github.com/opensds/opensds (development) # ./build/out/bin/osdsctl pool list --status available
WARNING: OPENSDS_ENDPOINT is not specified, use default(http://localhost:50040)
WARNING: Not found Env OPENSDS_AUTH_STRATEGY, use default(noauth)
+--------------------------------------+-------------------------+-------------+-----------+---------------+--------------+
| Id                                   | Name                    | Description | Status    | TotalCapacity | FreeCapacity |
+--------------------------------------+-------------------------+-------------+-----------+---------------+--------------+
| 47d06aef-afe0-5f34-90c1-66c355662a9b | opensds-volumes-default |             | available | 20            | 20           |
| 157e862c-7604-5d1b-b80a-37559a88e65d | opensds-files-default   |             | available | 20            | 20           |
+--------------------------------------+-------------------------+-------------+-----------+---------------+--------------+
root@node01:~/gopath/src/github.com/opensds/opensds (development) #
```

### CODE REVIEW

**Comment by wisererik** [comment]
please use Utils.Filter/Sort/Slice to handle the query data in memory.

**Comment by ThisIsClark** [comment]
Need more unit test

**Comment by ThisIsClark** [comment]
> Need more unit test

Done

**Comment by ThisIsClark** [comment]
> please use Utils.Filter/Sort/Slice to handle the query data in memory.

Done

File: pkg/db/drivers/etcd/etcd.go [hunk 1]
```diff
@@ -49,6 +49,28 @@ const (
 
 var validKey = []string{"limit", "offset", "sortDir", "sortKey"}
 
+const (
+	typeFileShares         string = "FileShares"
+	typeFileShareSnapshots string = "FileShareSnapshot"
```
[the remark this answers is not in the document]
Comment (ThisIsClark) [reply pkg/db/drivers/etcd/etcd.go]: Done

**Comment by ThisIsClark** [comment]
> please add test report

Done

**Pull request merged**